### PR TITLE
Implement RowwiseQuantizedSparseLengthsWeightedSum

### PR DESCRIPTION
**Pull request by jfix71**
*Description*: Implement RowwiseQuantizedSparseLengthsWeightedSum. Added support for both Interpreter and CPU backends.

*Testing*: Added unit tests for both RowwiseQuantizedSparseLengthsWeightedSum and RowwiseQuantizedSparseLengthsSum (implemented as the first but with weights as a float splat of 1.0).

*Documentation*: Added to Quantization.md

Related to https://github.com/pytorch/glow/issues/1698

### CODE REVIEW

File: lib/Quantization/Quantization.cpp [hunk 1]
```diff
@@ -420,29 +420,32 @@ class FunctionQuantizer : public FunctionConverter {
...
+          result.getType()->isQuantizedType()) {
+        auto *wq = llvm::dyn_cast<QuantizeNode>(weights.getNode());
+        // For RowwiseQuantizedFullyConnected, the weights need to be constant.
+        if (Constant *wc = llvm::dyn_cast<Constant>(wq->getInput())) {
+          auto *fcq = function_.createRowwiseQuantizedFullyConnected(
+              "rowwiseqfc", input, wc, bias, result.getType(),
+              /* transposeWeight */ true);
+          // Replace the usages of quantized FC node to
+          // RowwiseQuantizedFullyConnected Node.
+          result.replaceAllUsesOfWith(fcq->getResult());
```
Comment (beicy) [inline lib/Quantization/Quantization.cpp]: For the above diff,  it seems to me that only the code structure has been changed: 
`if (A) { if (B) {...} } => A =..; if (!A) continue; B=..; if (!B) continue; ...` ?  If so, I don't think it is related to this PR. Please let me know if I missed anything.

Comment (jfix71) [reply lib/Quantization/Quantization.cpp]: Ok, I've removed it.

File: include/glow/Graph/Graph.h [hunk 2]
```diff
@@ -565,6 +565,25 @@ class Function final : public Named {
...
                                  NodeValue indices, NodeValue lengths);
 
+  /// Create a node, performing SparseLengthsSum operation, using rowwise
+  /// quantization for the input data. Gathers slices of the outer-most
+  /// dimension of Data indexed by Indices vector, and then accumulates them
+  /// into len(Lengths) entries: first Lengths[0] slices are aggregated to
+  /// Result[0], next Lengths[1] slices are aggregated to Result[1],
+  /// etc. I.e. sum(Lengths) must be equal to len(Indices).
+  RowwiseQuantizedSparseLengthsWeightedSumNode *
+  createRowwiseQuantizedSparseLengthsSum(llvm::StringRef name, Tensor &data,
```
Comment (bertmaher) [inline include/glow/Graph/Graph.h]: Passing a `Tensor` for `data` seems unusual vs. a `NodeValue` or a `Constant`.  I guess it makes sense given its role though, so I don't feel too strongly.

Comment (jfix71) [reply include/glow/Graph/Graph.h]: Yeah, a `Constant`/`NodeValue` would essentially only be used as a container for `Tensor data`. I also thought it made sense to differentiate it from the other creator functions, to make it more clear that the data passed in is rowwise-quantized.

When we add in importer support for this Node which loads already-quantized `data` we can refactor/add a new creator method that takes a `NodeValue`/`Constant`.

**Comment by rdzhabarov** [comment]
We'd need to have C2 loader in place in order to close https://github.com/pytorch/glow/issues/1698

**Comment by jfix71** [comment]
I'm going to implement the C2 importer for this in a follow up PR.

**Pull request merged**